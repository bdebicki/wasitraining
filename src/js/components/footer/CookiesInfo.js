import { FOOTER, LINK } from '../../enums/elementHandlers';
import { COOKIES } from '../../enums/content';

export default class CookiesInfo {
	static showCookiesDetails(e) {
		e.preventDefault();

		console.log('cookies policy');
	}

	static renderCookiesLnk() {
		const link = document.createElement('a');

		link.textContent = 'here';
		link.href = '#cookies-policy';
		link.classList.add(LINK.BASIC_CLASS);
		link.addEventListener('click', CookiesInfo.showCookiesDetails, null);

		return link;
	}

	static render() {
		const cookiesInfoContainer = document.createElement('p');

		cookiesInfoContainer.textContent = COOKIES.PT1;
		cookiesInfoContainer.classList.add(FOOTER.COOKIES_CLASS);
		cookiesInfoContainer.appendChild(CookiesInfo.renderCookiesLnk());
		// textContent parse link to string, innerHtml += clear eventListener from link, append works correct
		cookiesInfoContainer.append(COOKIES.PT2);

		return cookiesInfoContainer;
	}
}
