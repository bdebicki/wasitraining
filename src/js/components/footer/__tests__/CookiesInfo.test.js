import CookiesInfo from '../CookiesInfo';

describe('test cookies info section', () => {
  it('render cookies section', () => {
    const cookies = CookiesInfo.render();

    expect(cookies).toMatchSnapshot();
  });
});
