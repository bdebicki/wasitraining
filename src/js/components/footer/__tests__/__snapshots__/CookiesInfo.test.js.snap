// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test cookies info section render cookies section 1`] = `
<p
	class="cookiesSection"
>
	yep. this site uses cookies. click
	<a
		class="link"
		href="#cookies-policy"
	>
		here
	</a>
	 for more.
</p>
`;
